### PR TITLE
Setup Github Actions for Antenna

### PR DESCRIPTION
**Pull request by blaumeiser-at-bosch**
Signed-off-by: Lars Geyer-Blaumeiser <lars.geyer-blaumeiser@bosch-si.com>

### Request Reviewer
@sschuberth @bs-ondem 

### Type of Change

*Type of change*:  CI

### How Has This Been Tested?
Run build script in PR build and Travis build as well

### Checklist
- [ ] All related issues are referenced in commit messages
- [ ] I have updated the documentation accordingly to my changes

### CODE REVIEW

File: .github/workflows/pull_request_build.yml [hunk 1]
```diff
@@ -6,12 +6,11 @@
 # http://www.eclipse.org/legal/epl-v20.html
 #
 # SPDX-License-Identifier: EPL-2.0
-name: Antenna Push Build
+name: Antenna Pull Request Build
```
Comment (sschuberth) [inline .github/workflows/pull_request_build.yml]: Commit message:
- s/travsi/Travis/
- s/gitub/GitHub/

In general, from readin the commit message I would have expected a `.travis.yml` being touched / removed in favor of a GitHub actions YAML, but that's not what seems to be the case in the strict sense. In fact, nothing Travis-related seems to happen in this commit at all.

File: .github/workflows/pull_request_build.yml [hunk 2]
```diff
@@ -33,10 +32,49 @@ jobs:
...
+      run: bash ./modules/p2/prepareDependenciesForP2.sh
+
     - name: Build with Maven
-      run: mvn -B package --file pom.xml
+      run: mvn -B clean install -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+
+    - name: Run SCA Analysis
+      run: mvn verify -DskipTests pmd:pmd checkstyle:checkstyle-aggregate spotbugs:check
+
+#    - name: Run SonarCloud Analysis
```
Comment (sschuberth) [inline .github/workflows/pull_request_build.yml]: I'd prefer to not commit any out-commented code. Also see e.g. https://kentcdodds.com/blog/please-dont-commit-commented-out-code.

File: .github/workflows/pull_request_build.yml [hunk 3]
```diff
@@ -41,15 +41,15 @@ jobs:
       run: bash ./modules/p2/prepareDependenciesForP2.sh
 
     - name: Build with Maven
-      run: mvn -B clean install -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+      run: mvn -B clean install -Pcoverage -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
```
Comment (sschuberth) [inline .github/workflows/pull_request_build.yml]: The commit message should mention in what context SonarCloud is enabled (in the context of GitHub actions), and what's the benefit of doing so (static code analysis).

File: .github/workflows/pull_request_build.yml [hunk 4]
```diff
@@ -51,32 +51,32 @@ jobs:
       env:
         SONAR_TOKEN: "e5dbb8f62a0b8bf02dbbcf4c9fd745c31b63f4b3"
 
-#    - name: Run Example Project Test with JDK8
-#      run: |
-#        cd example-projects/example-project
-#        mvn clean package --> jsonld issue
+    - name: Run Example Project Test with JDK8
```
Comment (sschuberth) [inline .github/workflows/pull_request_build.yml]: Again the commit message lacks the context, so it's unclear whether tests as added to the code base, or just to the CI scripts (i.e. their are not "added" but "run").

File: assembly/spdx-license-knowledge-base/pom.xml [hunk 5]
```diff
@@ -61,8 +61,17 @@
                     <groupId>log4j</groupId>
                     <artifactId>log4j</artifactId>
                 </exclusion>
+                <exclusion>
```
Comment (sschuberth) [inline assembly/spdx-license-knowledge-base/pom.xml]: These POM changes deserve some extra justification / explanation in the commit message body, i.e. why are they required *now* but not before?

**Comment by blaumeiser-at-bosch** [comment]
@sschuberth: Commit messages enhanced. Please give feedback

File: .github/workflows/pull_request_build.yml (hunk 3 above)
Comment (sschuberth) [reply .github/workflows/pull_request_build.yml]: Some further commit message nits to address *optionally*:
- "Enable the SonarCloud step in the GH action script" should simply become the title.
- For the common reader it's unclear what "With GH actions this is possible for PR builds as well" refers to, i.e. what the problem with Travis (not PRs in general) is. A link to https://docs.travis-ci.com/user/environment-variables/#defining-encrypted-variables-in-travisyml would help to explain that.

File: .github/workflows/pull_request_build.yml (hunk 4 above)
Comment (sschuberth) [reply .github/workflows/pull_request_build.yml]: The commit message mysteriously talks about "an issue with JsonId", but it should explain in this context what exactly the problem with JsonId was.

**Pull request merged**